src/Microsoft.Identity.Web/Resource/RegisterValidAudience.cs

```diff
@@ -61,7 +61,12 @@ internal class RegisterValidAudience
             SecurityToken securityToken,
             TokenValidationParameters validationParameters)
         {
-            JwtSecurityToken token = securityToken as JwtSecurityToken;
+            JwtSecurityToken? token = securityToken as JwtSecurityToken;
+            if (token == null)
+            {
+                throw new SecurityTokenValidationException("Token is not JWT token.");
+            }
```

Comment:
is that the right error message? when would the casting fail in this case and result in a null token value? the message should have some customer value, so how can they resolve or understand the issue to fix it.

Reply:
I am not sure if it's always `JwtSecurityToken`. Is it? If it is, I can just explicitly cast it.

Reply:
might be something to look into. the error message is misleading.